Modules/selectmodule.c

```diff
@@ -532,11 +532,12 @@ poll_poll(pollObject *self, PyObject *args)
         return NULL;
     }
 
+    deadline = 0;   /* initialize to prevent gcc warning */
+
     /* Check values for timeout */
     if (timeout_obj == NULL || timeout_obj == Py_None) {
```

Comment:
Looking at this again, we can get rid of this if branch if we move out default values to variable declarations:

```c
_PyTime_t timeout = -1, ms = -1, deadline = 0;
```

Then we can get rid of the following comment too:

```c
/* Check values for timeout */
```